### PR TITLE
fix(postman): prevent infinite recursion in variable substitution

### PR DESCRIPTION
**Pull request by dustin-decker**
Add recursion depth limit and self-reference detection to the buildSubstitution function to prevent hanging when processing complex variable patterns. Includes comprehensive tests for edge cases.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

File: pkg/sources/postman/substitution.go [hunk 1]
```diff
@@ -63,26 +76,62 @@ func (s *Source) buildSubstituteSet(metadata Metadata, data string) []string {
 	return ret
 }
 
-func (s *Source) buildSubstitution(data string, metadata Metadata, combos *map[string]struct{}) {
+// buildSubstitution performs variable substitution with a maximum recursion depth
+// buildSubstitution performs variable substitution with a maximum recursion depth
```
Comment (casey-tran) [inline pkg/sources/postman/substitution.go:80]: Duplicate comment.

File: pkg/sources/postman/substitution.go [hunk 2]
```diff
@@ -63,26 +76,62 @@ func (s *Source) buildSubstituteSet(metadata Metadata, data string) []string {
...
+// maxDepth is an optional parameter to specify the maximum recursion depth
+// if not provided, DefaultMaxRecursionDepth will be used
+func (s *Source) buildSubstitution(data string, metadata Metadata, combos *map[string]struct{}, depth int, maxDepth ...int) {
```
Comment (casey-tran) [inline pkg/sources/postman/substitution.go:83]: Since the initial call of `buildSubstitution` is inside of `buildSubstituteSet`, wouldn't `maxDepth` always have at the minimum the default value of `DefaultMaxRecursionDepth`? So there is no need to make `maxDepth` a trailing argument since for this function it isn't really optional?

File: pkg/sources/postman/substitution.go [hunk 3]
```diff
@@ -63,26 +76,62 @@ func (s *Source) buildSubstituteSet(metadata Metadata, data string) []string {
...
+	maxRecursionDepth := DefaultMaxRecursionDepth
+	if len(maxDepth) > 0 && maxDepth[0] > 0 {
+		maxRecursionDepth = maxDepth[0]
+	}
+
+	// Limit recursion depth to prevent stack overflow
+	if depth > maxRecursionDepth {
+		(*combos)[data] = struct{}{}
+		return
+	}
```
Comment (casey-tran) [inline pkg/sources/postman/substitution.go:94]: Assuming from the previous comment about `maxDepth` always holding a value from the initial call of `buildSubstitution` holds true, would the assigning of values to `maxRecursionDepth` and using `maxRecursionDepth` be redundant if you could just check `if depth > maxDepth `?

File: pkg/sources/postman/substitution.go [hunk 4]
```diff
@@ -63,26 +76,62 @@ func (s *Source) buildSubstituteSet(metadata Metadata, data string) []string {
...
+
+			// Use the actual value for substitution, not just the stripped version
+			d := strings.ReplaceAll(data, match, slice.value)
 
-			d := strings.ReplaceAll(data, match, removedBrackets)
-			s.buildSubstitution(d, metadata, combos)
+			// Only mark substitution as made if we actually changed something
+			if d != data {
+				substitutionMade = true
+				s.buildSubstitution(d, metadata, combos, depth+1, maxRecursionDepth)
```
Comment (casey-tran) [inline pkg/sources/postman/substitution.go:128]: Building from my other comments, I'm wondering if `maxRecursionDepth` can't be replaced by `maxDepth` here.

File: pkg/sources/postman/substitution.go (hunk 2 above)
Comment (dustin-decker) [reply pkg/sources/postman/substitution.go]: I've made max recursion depth a required parameter to simplify this.

File: pkg/sources/postman/substitution.go (hunk 4 above)
Comment (dustin-decker) [reply pkg/sources/postman/substitution.go:114]: I've made max recursion depth a required parameter to simplify this.

File: pkg/sources/postman/substitution.go (hunk 3 above)
Comment (dustin-decker) [reply pkg/sources/postman/substitution.go]: I've made max recursion depth a required parameter to simplify this.

**Review by casey-tran — APPROVED** [review]
Thanks for looking into this Dustin.
